rules/gdal.json

```diff
@@ -4,6 +4,24 @@
     "\\blibgdal\\b"
   ],
   "dependencies": [
+    {
+      "packages": [
+        "libgdal-dev",
+        "gdal-bin"
+      ],
+      "pre_install": [
+        { "command": "apt-get install -y software-properties-common" },
+        { "command": "add-apt-repository ppa:ubuntugis/ppa" },
```

Comment:
Shouldn't geos make use of the same PPA?

Reply:
Oh hmm, I'm not sure. libgeos is 3.5.0 in the official xenial repo, but 3.5.1 in the UbuntuGIS repo. Would a difference in the patch versions be problematic if a package was built with 3.5.0, but run on a system with 3.5.1?

Reply:
Well, doesn't it become an ordering issue?
If you install `rgdal` before `sf`, then you ending with the UbuntuGIS repo, and therefore libgeos is installed from there.
But if you install `sf` first, then libgeos is installed from the base apt repo, and then you install `rgdal` and it may actually want to update libgeos-dev from the UbuntuGIS repo.
It just feels like you end up with potentially different things based on ordering, and they may work, but they may not.

Reply:
That's true. We don't have this specific problem with `rgdal` and `sf` in RSPM because we run all pre-install steps before installing any packages, and `sf`'s libgeos comes from UbuntuGIS. But there's no dictated way to interpret these rules, so anyone else could run into ordering issues like this.

Another example would be with packages like `rgeos`, which only needs libgeos. If you install `rgeos` first and `sf` second, you'd compile `rgeos` with libgeos 3.5.0 from the base repo, but end up with libgeos 3.5.1 from UbuntuGIS. Then when you run `rgeos`, the library loads but there's a warning message:
```r
> library(rgeos)
Loading required package: sp
rgeos version: 0.5-1, (SVN revision 614)
GEOS runtime version: 3.5.1-CAPI-1.9.1 
Linking to sp version: 1.3-1 
Polygon checking: TRUE 

Warning message:
In fun(libname, pkgname) : rgeos: versions of GEOS runtime 3.5.1-CAPI-1.9.1
and GEOS at installation 3.5.0-CAPI-1.9.0differ
```

I agree that geos should use the same PPA now, and I'm also wondering if there are any other cases like this.. We have a bunch of rules using EPEL, and a few others using various 3rd party repos on openSUSE/SLES.